src/aliceVision/image/dcp.cpp

```diff
@@ -0,0 +1,1522 @@
+    else if(type == TagType::T_ASCII)
+    {
+        snprintf(buffer, size, "%.64s", v_value.data() + ofs);
+        return;
+    }
+
+    size_t maxcount = std::min<size_t>(datasize, 10);
+
+    buffer[0] = 0;
+
+    for(int i = 0; i < std::min<int>(maxcount, datasize * getTypeSize(type) - ofs); i++)
+    {
+        std::size_t len = strlen(buffer);
+
+        if(i > 0 && size - len > 2)
+        {
+            strcat(buffer, ", ");
+            len += 2;
+        }
+
+        char* b = buffer + len;
+
+        switch(type)
+        {
+            case TagType::T_UNDEFINED:
+            case TagType::T_BYTE:
+                snprintf(b, size - len, "%d", v_value[i + ofs]);
+                break;
+
+            case TagType::T_SSHORT:
+                snprintf(b, size - len, "%d", toInt(2 * i + ofs));
+                break;
+
+            case TagType::T_SHORT:
+                snprintf(b, size - len, "%u", toInt(2 * i + ofs));
+                break;
+
+            case TagType::T_SLONG:
+                snprintf(b, size - len, "%d", toInt(4 * i + ofs));
+                break;
+
+            case TagType::T_LONG:
+                snprintf(b, size - len, "%u", toInt(4 * i + ofs));
+                break;
+
+            case TagType::T_SRATIONAL:
+                snprintf(b, size - len, "%d/%d", (int)sget4(v_value.data() + 8 * i + ofs, order),
+                         (int)sget4(v_value.data() + 8 * i + ofs + 4, order));
+                break;
+
+            case TagType::T_RATIONAL:
+                snprintf(b, size - len, "%u/%u", (uint32_t)sget4(v_value.data() + 8 * i + ofs, order),
+                         (uint32_t)sget4(v_value.data() + 8 * i + ofs + 4, order));
+                break;
+
+            case TagType::T_FLOAT:
+                snprintf(b, size - len, "%g", toDouble(8 * i + ofs));
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    if(datasize > maxcount && size - strlen(buffer) > 3)
+    {
+        strcat(buffer, "...");
+    }
+}
+
+std::string Tag::valueToString() const
+{
+    char buffer[1024];
+    toString(buffer, sizeof(buffer));
+    return buffer;
+}
+
+Tag* findTag(TagKey tagID, std::vector<Tag>& v_Tags)
+{
+    size_t idx = 0;
+    while(idx < v_Tags.size() && (v_Tags[idx].tagID != (unsigned short)tagID))
+    {
+        idx++;
+    }
+    return (idx == v_Tags.size()) ? nullptr : &v_Tags[idx];
+}
+
+void SplineToneCurve::Set(const std::vector<double>& v_xy)
+{
+    std::vector<double> v_x;
+    std::vector<double> v_y;
+    std::vector<double> v_ypp;
+
+    const size_t N = v_xy.size() / 2;
+
+    for(int i = 0; i < N; ++i)
+    {
+        v_x.push_back(v_xy[2 * i]);
+        v_y.push_back(v_xy[2 * i + 1]);
+        v_ypp.push_back(0.0);
+    }
+    std::vector<double> v_u(N - 1, 0.0);
+
+    v_ypp[0] = v_u[0] = 0.0; /* Natural spline */
+
+    for(int i = 1; i < N - 1; ++i)
+    {
+        const double sig = (v_x[i] - v_x[i - 1]) / (v_x[i + 1] - v_x[i - 1]);
```

Comment:
I think it makes sense to add a comment explaining maths behind the computation here, maybe a link to wikipedia page or documentation of DCP standard.